src/decoder/stream.rs

```diff
@@ -1668,8 +1636,8 @@ impl StreamingDecoder {
                 // TODO: Calculate **exact** IDAT size for interlaced images.
                 raw_row_len = raw_row_len.saturating_mul(2);
             }
-            self.inflater
-                .set_max_total_output((height as usize).saturating_mul(raw_row_len));
+            // self.inflater
+            //     .set_max_total_output((height as usize).saturating_mul(raw_row_len));
```

Comment:
Should add back max total output calculation and use it to avoid overallocating for unfiltering buffer.